server/src/UseCases/TranscriptUseCases/transcriptUploadInteractor.js

```diff
@@ -0,0 +1,43 @@
+const processTranscriptInteractor = require('./intentIdentifierInteractor')
+
+const uploadTranscriptInteractor = async (user, file, filename) => {
+    const json = JSON.parse(file)
+
+    let allCurrentIntents = new Map()
+
+    if (user.intents !== undefined) {
+        allCurrentIntents = new Map(Object.entries(user.intents));
+    }
+
+    //single transcript processing
+    let intentsForThisFile = processTranscriptInteractor(new Map(), [json])
+
+    //multiple transcript processing
+    allCurrentIntents = processTranscriptInteractor(allCurrentIntents, [json])
+
+    // save transcript
+    try {
+        const obj = {}
+        obj[filename] = file
+        obj["intents"] = intentsForThisFile
+        user.transcripts = user.transcripts.concat(obj)
+        await user.save()
+    }
+    catch (e) {
+        throw new Error("Error in saving transcripts", e)
+    }
+
+    // save intents
+    try {
+        user.intents = allCurrentIntents
+        await user.save()
+    }
+    catch (e) {
+        throw new Error("Error in saving intents", e)
+    }
```

Comment:
Great job adding error catching when saving the transcript. 
One thing I've encountered is when I upload two transcripts of the same name then an _Axios Error: Network Error_ is displayed, try it yourself and then see if you can catch this error too.

Reply:
yes I've noticed the same error as well, good catch! Let me fix this by the end of the day.